boot/espressif/CMakeLists.txt

```diff
@@ -7,66 +7,83 @@ cmake_policy(SET CMP0109 NEW)
 
 include(${CMAKE_CURRENT_LIST_DIR}/tools/utils.cmake)
 
-set(MCUBOOT_ROOT_DIR ${CMAKE_CURRENT_LIST_DIR}/../..)
-set(ESPRESSIF_PORT_DIR ${MCUBOOT_ROOT_DIR}/boot/espressif)
+# SYSBUILD variable indicates that the build was triggered by Zephyr sysbuild
+if(SYSBUILD)
+    message(STATUS "Building MCUboot for Zephyr OS -- ESPRESSIF PORT")
 
-set(APP_NAME mcuboot_${MCUBOOT_TARGET})
-set(APP_EXECUTABLE ${APP_NAME}.elf)
-set(EXPECTED_IDF_HAL_VERSION "5.1.4")
+    find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
```

Comment:
bit confused here, if sysbuild is set then this should already be from zephyr scope, why is this finding zephyr when it is running from zephyr?

Reply:
This is needed because the project is being included as an external Zephyr project from sysbuild (https://github.com/zephyrproject-rtos/zephyr/pull/87710).
MCUboot Zephyr port does the same to include the Zephyr environment.